Sources/CoffeeBot/Models/Coffee.swift

```diff
@@ -0,0 +1,18 @@
+//
+//  Coffee.swift
+//  Coffee
+//
+
+internal struct Coffee {
+    let amount : Int
+    let type : CoffeeType
```

Comment:
Sure, https://github.com/s/SwiftBot/blob/master/Sources/ChatProviders/Facebook/MessengerWebhook.swift#L72
or in your case:
```swift
internal struct Coffee {
    let amount : Int
    enum `Type`: String {
        case espresso
        case capuchino
    }
    let type: Type
}
```
Then you can extend your type with this:
```swift
extension Coffee.Type {
    func ...
}
```